iati/core/default.py

```diff
@@ -105,35 +105,65 @@ def codelists(version=None, bypass_cache=False):
 
 
 def schemas(bypass_cache=False):
-    """Locate the default Schemas.
+    """Locate all the default IATI Schemas and return them within a dictionary.
 
     Args:
         bypass_cache (bool): Whether the cache should be bypassed, instead reloading data from disk even if it's already been loaded.
 
     Returns:
-        dict: A dictionary containing all the Schemas for versions of the Standard. The version of the Standard is the key. An iati.core.Schema() is each value.
+        dict: A dictionary containing all the Schemas for versions of the Standard. The name of the schema the key. An iati.core.Schema() is each value.
 
     Warning:
         The `bypass_cache` parameter could potentially be implemented in a cleaner manner. It also shouldn't really exist until a clear use-case is defined - changes elsewhere in the library may make it redundant.
 
     Todo:
-        Allow creation of Schemas by XML rather than name.
-
-        Handle the difference between Organisation and Activity Schemas.
-
         Consider the Schema that defines the format of Codelists.
 
         Test a cache bypass where data is updated.
 
         Load the Schemas.
 
-    """
-    paths = iati.core.resources.find_all_schema_paths()
+        Needs to handle multiple versions of the Schemas. Versions could perhaps be placed within a nested dictionary, with the version number as the key.
 
-    for path in paths:
-        name = path.split(os.sep).pop()[:-len(iati.core.resources.FILE_SCHEMA_EXTENSION)]
-        if (name not in _SCHEMAS.keys()) or bypass_cache:
-            schema = iati.core.Schema(name)
-            _SCHEMAS[name] = schema
+    """
+    schema_paths_by_type = {
```

Comment:
Potentially split into 3 functions:

* `activity_schemas()`
* `org_schemas()`
* `schemas()` (that then calls the others to obtain the data)